### PR TITLE
Externalize JavaScript: edit user & data dictionary

### PR DESCRIPTION
**Pull request by orangejenny**
@millerdev

### CODE REVIEW

**Comment by dimagimon** [comment]
There are 24 errors:

* corehq/apps/users/static/users/js/edit_commcare_user.js, line 16 - Missing trailing comma. (comma-dangle)
* corehq/apps/users/static/users/js/edit_commcare_user.js, line 29 - 'e' is defined but never used (no-unused-vars)
* corehq/apps/users/static/users/js/edit_commcare_user.js, line 42 - Expected '===' and instead saw '=='. (eqeqeq)
* corehq/apps/users/static/users/js/edit_commcare_user.js, line 44 - 'alert_user' is not defined. (no-undef)
* corehq/apps/users/static/users/js/edit_commcare_user.js, line 47 - 'message' is not defined. (no-undef)
* corehq/apps/users/static/users/js/edit_commcare_user.js, line 50 - 'message' is not defined. (no-undef)
* corehq/apps/users/static/users/js/edit_commcare_user.js, line 54 - 'alert_user' is not defined. (no-undef)
'message' is not defined. (no-undef)
* corehq/apps/users/static/users/js/edit_commcare_user.js, line 58 - 'response' is defined but never used (no-unused-vars)
* corehq/apps/users/static/users/js/edit_commcare_user.js, line 59 - Unexpected console statement. (no-console)
* corehq/apps/users/static/users/js/edit_commcare_user.js, line 60 - Missing trailing comma. (comma-dangle)
* corehq/apps/users/static/users/js/edit_commcare_user.js, line 65 - Move function declaration to function body root. (no-inner-declarations)
* corehq/apps/users/static/users/js/edit_commcare_user.js, line 66 - Expected indentation of 12 space characters but found 13. (indent)
* corehq/apps/users/static/users/js/edit_commcare_user.js, line 67 - Expected indentation of 12 space characters but found 13. (indent)
* corehq/apps/users/static/users/js/edit_commcare_user.js, line 68 - Expected indentation of 12 space characters but found 13. (indent)
* corehq/apps/users/static/users/js/edit_commcare_user.js, line 69 - Expected indentation of 12 space characters but found 13. (indent)
* corehq/apps/users/static/users/js/edit_commcare_user.js, line 73 - Expected indentation of 12 space characters but found 13. (indent)
* corehq/apps/users/static/users/js/edit_commcare_user.js, line 78 - Missing semicolon. (semi)
* corehq/apps/users/static/users/js/edit_commcare_user.js, line 79 - Expected indentation of 8 space characters but found 9. (indent)
* corehq/apps/users/static/users/js/edit_commcare_user.js, line 80 - Expected indentation of 8 space characters but found 9. (indent)
* corehq/apps/users/static/users/js/edit_commcare_user.js, line 90 - Missing trailing comma. (comma-dangle)
* corehq/apps/users/static/users/js/edit_commcare_user.js, line 99 - Expected indentation of 4 space characters but found 8. (indent)
* corehq/apps/users/static/users/js/edit_commcare_user.js, line 101 - 'django' is not defined. (no-undef)
* corehq/apps/users/static/users/js/edit_commcare_user.js, line 102 - 'django' is not defined. (no-undef)
* corehq/apps/users/static/users/js/edit_commcare_user.js, line 103 - 'django' is not defined. (no-undef)

**Comment by dimagimon** [comment]
There are 6 errors:

* corehq/apps/users/static/users/js/edit_commcare_user.js, line 43 - 'alert_user' is not defined. (no-undef)
* corehq/apps/users/static/users/js/edit_commcare_user.js, line 51 - 'alert_user' is not defined. (no-undef)
* corehq/apps/users/static/users/js/edit_commcare_user.js, line 58 - Move function declaration to function body root. (no-inner-declarations)
* corehq/apps/users/static/users/js/edit_commcare_user.js, line 71 - Missing semicolon. (semi)
* corehq/apps/users/static/users/js/edit_commcare_user.js, line 74 - Expected indentation of 12 space characters but found 11. (indent)
* corehq/apps/users/static/users/js/edit_commcare_user.js, line 92 - Expected indentation of 4 space characters but found 8. (indent)

File: corehq/apps/users/static/users/js/edit_commcare_user.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+    $('#reset-password-form').submit(function(){
+        $(this).ajaxSubmit({
+            url: $(this).attr('action'),
+            type: 'POST',
+            dataType: 'json',
+            success: function (response, status, xhr, form) {
+                form.find('#user-password').html(response.formHTML);
+                if (response.status === "OK") {
+                    alert_user(gettext("Password changed successfully"), 'success');
```
Comment (dimagimon) [inline corehq/apps/users/static/users/js/edit_commcare_user.js]: 'alert_user' is not defined. (no-undef)

File: corehq/apps/users/static/users/js/edit_commcare_user.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+                    ga_track_event("Edit Mobile Worker", "Reset password", couch_user_id);
+                } else {
+                    var message = gettext('Password was not changed ');
+                    if (initial_page_data('hide_password_feedback')) {
+                        message += gettext("Password Requirements: 1 special character, " +
+                                          "1 number, 1 capital letter, minimum length of 8 characters.");
+                    }
+                    alert_user(message, 'danger');
```
Comment (dimagimon) [inline corehq/apps/users/static/users/js/edit_commcare_user.js]: 'alert_user' is not defined. (no-undef)

File: corehq/apps/users/static/users/js/edit_commcare_user.js [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+                }
+            },
+        });
+        return false;
+    });
+    if (!initial_page_data('is_currently_logged_in_user')) {
+        function DeleteUserButtonModel() {
```
Comment (dimagimon) [inline corehq/apps/users/static/users/js/edit_commcare_user.js]: Move function declaration to function body root. (no-inner-declarations)

File: corehq/apps/users/static/users/js/edit_commcare_user.js [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+                },
+            });
+            return false;
+        });
+
+    }
+
+    // Groups form
+    var multiselect_utils = hqImport('style/js/multiselect_utils');
+        multiselect_utils.createFullMultiselectWidget(
```
Comment (dimagimon) [inline corehq/apps/users/static/users/js/edit_commcare_user.js]: Expected indentation of 4 space characters but found 8. (indent)

File: corehq/apps/users/static/users/js/edit_commcare_user.js [hunk 5]
```diff
@@ -0,0 +1,113 @@
+$(function () {
```
Comment (millerdev) [inline corehq/apps/users/static/users/js/edit_commcare_user.js]: So many things begging to be refactored in here... Good for you for being disciplined and only doing one thing at a time. I was about to suggest `hqDefine` given the size of this document ready handler, but it looks like that's more than a simple endeavor.

**Review by millerdev — APPROVED** [review]

**Pull request merged**